astroid/inference.py

```diff
@@ -1063,20 +1077,26 @@ def _cached_generator(func, instance, args, kwargs, _cache={}):  # noqa: B006
 # are mutated with a new instance of the property. This is why we cache the result
 # of the function's inference.
 @_cached_generator
-def infer_functiondef(self, context=None):
+def infer_functiondef(
+    self: nodes.FunctionDef, context: Optional[InferenceContext] = None
+) -> Generator[
+    Union["astroid_objects.Property", nodes.FunctionDef],
+    None,
+    Dict[str, Union[nodes.FunctionDef, InferenceContext, None]],
+]:
     if not self.decorators or not bases._is_property(self):
         yield self
         return dict(node=self, context=context)
 
     prop_func = objects.Property(
         function=self,
         name=self.name,
-        doc=self.doc,
+        doc=self.doc_node.value if self.doc_node else None,
```

Comment:
```suggestion
```
If `postinit` sets `doc`, this shouldn't be necessary. See https://github.com/PyCQA/astroid/pull/1434#pullrequestreview-899198368

Reply:
See my comment there.